pkg/scheduler/framework/types.go

```diff
@@ -705,21 +705,26 @@ func podWithRequiredAntiAffinity(p *v1.Pod) bool {
 		len(affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution) != 0
 }
 
-func removeFromSlice(s []*PodInfo, k string) []*PodInfo {
+func removeFromSlice(s []*PodInfo, k string) ([]*PodInfo, bool) {
+	var removed bool
 	for i := range s {
-		k2, err := GetPodKey(s[i].Pod)
+		tmpKey, err := GetPodKey(s[i].Pod)
 		if err != nil {
 			klog.ErrorS(err, "Cannot get pod key", "pod", klog.KObj(s[i].Pod))
 			continue
 		}
-		if k == k2 {
+		if k == tmpKey {
 			// delete the element
 			s[i] = s[len(s)-1]
 			s = s[:len(s)-1]
+			removed = true
 			break
 		}
 	}
-	return s
+	if len(s) == 0 {
```

Comment:
add a comment for this?
```suggestion
        // resets the slices to nil so that we can do DeepEqual in unit tests
	if len(s) == 0 {
```

Reply:
Make sense, updated, I forgot the comment.